nameko/timer.py

```diff
@@ -56,27 +56,28 @@ def kill(self):
 
     def _run(self):
         """ Runs the interval loop. """
-        sleep_time = 0 if self.eager else self.interval
 
+        def get_next_interval():
+            start = time.time()
+            count = 0
+            while True:
+                count += 1
```

Comment:
Would be slightly terser to use an itertools counter here:

``` python
start = time.time()
for count in itertools.count(start=1):
    yield max(start + count * self.interval - time.time(), 0)
```

With this, `eager` also approximates `start=0`, which may or may not be a nicer way to implement it. Can't quite decide.